searchlib/src/vespa/searchlib/attribute/attribute_blueprint_factory.cpp

```diff
@@ -592,11 +602,43 @@ class CreateBlueprintVisitor : public CreateBlueprintVisitorHelper
             createShallowWeightedSet(bp, n, _field, _attr.isIntegerType());
         }
     }
-    void visit(query::NearestNeighborTerm &n) override {
-        (void) n;
-        // TODO (geirst): implement
+    void fail_nearest_neighbor_term(query::NearestNeighborTerm&n, const vespalib::string& error_msg) {
+        LOG(warning, "NearestNeighborTerm(%s, %s): %s. Returning empty blueprint",
+            _field.getName().c_str(), n.get_query_tensor_name().c_str(), error_msg.c_str());
         setResult(std::make_unique<queryeval::EmptyBlueprint>(_field));
     }
+    void visit(query::NearestNeighborTerm &n) override {
+        if (_attr.asTensorAttribute() == nullptr) {
+            return fail_nearest_neighbor_term(n, "Attribute is not a tensor");
+        }
+        const auto* dense_attr_tensor = dynamic_cast<const DenseTensorAttribute*>(_attr.asTensorAttribute());
+        if (dense_attr_tensor == nullptr) {
+            return fail_nearest_neighbor_term(n, make_string("Attribute is not a dense tensor (type=%s)",
+                                                             _attr.asTensorAttribute()->getTensorType().to_spec().c_str()));
+        }
+        if (dense_attr_tensor->getTensorType().dimensions().size() != 1) {
+            return fail_nearest_neighbor_term(n, make_string("Attribute tensor type (%s) is not of order 1",
+                                                             dense_attr_tensor->getTensorType().to_spec().c_str()));
+        }
+        auto query_tensor = getRequestContext().get_query_tensor(n.get_query_tensor_name());
+        if (query_tensor.get() == nullptr) {
+            return fail_nearest_neighbor_term(n, "Query tensor was not found in request context");
+        }
+        auto* dense_query_tensor = dynamic_cast<DenseTensorView*>(query_tensor.get());
+        if (dense_query_tensor == nullptr) {
+            return fail_nearest_neighbor_term(n, make_string("Query tensor is not a dense tensor (type=%s)",
+                                                             query_tensor->type().to_spec().c_str()));
+        }
+        if (dense_attr_tensor->getTensorType() != dense_query_tensor->type()) {
+            // TODO: consider allowing different data types (float vs double).
```

Comment:
comparing type().dimensions() here will allow for float/double cells

Reply:
Thanks! Talked with @arnej27959, and I think we will defer adding support for this until we have something working end to end.